packages/qdrant-loader/src/qdrant_loader/core/state/state_manager.py

```
@@ -264,6 +264,99 @@ async def mark_document_deleted(
+        if document_ids_to_delete:
+            try:
+                await qdrant_manager.delete_points_by_document_id(
+                    document_ids_to_delete
+                )
+                self.logger.info(
+                    f"Deleted {len(document_ids_to_delete)} documents' points from Qdrant"
+                )
+            except Exception as e:
+                # Qdrant delete failed, but state is already committed
+                # Log the failure; the operation should be re-enqueued for retry (idempotent)
+                self.logger.error(
```

Ruff (0.15.14)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)